politeiad/backendv2/tstorebe/plugins/pi/hooks_test.go

```diff
@@ -449,17 +791,82 @@ func filesForProposal(t *testing.T, files ...backend.File) []backend.File {
 	return fs
 }
 
-// filesWithNoProposalName returns the backend files for a valid proposal,
+// filesWithProposalName returns the backend files for a valid proposal,
 // using the provided name as the proposal name. The returned files only
 // include the files required by the pi plugin API. No attachment files are
 // included.
 func filesWithProposalName(t *testing.T, name string) []backend.File {
 	t.Helper()
 
+	pm := fileProposalMetadata(t, &pi.ProposalMetadata{
+		Name: name,
+	})
+	return []backend.File{
+		fileProposalIndex(),
+		pm,
+	}
+}
+
+// filesWithProposalAmount returns the backend files for a valid proposal,
+// using the provided amount as the amount. The returned files only
+// include the files required by the pi plugin API. No attachment files are
+// included.
+func filesWithProposalAmount(t *testing.T, amount uint64) []backend.File {
```

Comment:
It occured to me that the refactor of the `fileProposalMetadata()` function in this PR allows us to get rid of all of these `filesWithProposal...` functions.

```
			filesWithProposalAmount(t, amountTooSmall),
```

Should be

```
			filesForProposal(t, &pi.ProposalMetadata{
				Amount: amountTooSmall,
			}),
```

Where `filesForProposal()` is:

```
func filesForProposal(t *testing.T, pm *pi.ProposalMetadata, files ...backend.File) []backend.File {
	t.Helper()

	fs := []backend.File{
		fileProposalIndex(),
		fileProposalMetadata(t, pm),
	}
	fs = append(fs, files...)

	return fs
}
```

Can you update this throughout the tests and remove all these extra functions?

Reply:
Like that, sure thing.